File: app/src/main/java/com/adut/pasar/app/MyApplication.java
package com.adut.pasar.app;

import android.app.Application;

import com.adut.pasar.app.di.AppComponent;
import com.adut.pasar.app.di.AppModule;
import com.adut.pasar.app.di.DaggerAppComponent;

public class MyApplication extends Application {

    public static AppComponent appComponent;
    private static MyApplication mInstance;

    public static MyApplication getInstance(){
        return mInstance;
    }


    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
        initDI();
    }

    private void initDI() {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }
}

File: app/src/main/java/com/adut/pasar/app/MyApplication.kt
package com.adut.pasar.app

import android.app.Application
import com.adut.pasar.app.di.AppComponent
import com.adut.pasar.app.di.DaggerAppComponent

class MyApplication: Application() {



    companion object {
        @JvmStatic lateinit var mInstance : MyApplication
        @JvmStatic lateinit var appComponent: AppComponent
        @Synchronized
        fun getInstance(): MyApplication {
            return MyApplication.mInstance
        }
    }

    init {
        mInstance = this
    }

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.create()
        appComponent.inject(this)
    }


}